### PR TITLE
feat: build nvidia images in main repo

### PR DESCRIPTION
**Pull request by bsherman**
This adds the nvidia build scripts to the repo and reconfigures `Containerfile` to build both `FOO-main` and `FOO-nvidia` images per matrix provided build-args. Workflow matrix was modified to support this use case, including changes to variables and tag generation.

Reviewers: please double-check the generated tags look correct per image. I did review and put some effort into this, but would appreciate extra eyes.

Changes which were not technically required:
- "main" `build.sh`, `post-install.sh`, `packages.json` renamed (with `main-` prefix) to make things more consistent and easier to follow for future
- switched to COPY instead of ADD in `Containerfile` for compliance with best practices
- added retry logic to curl in `github-release-install.sh` to avoid some observed intermittent errors

### CODE REVIEW

**Comment by bsherman** [comment]
After merging, we must disable actions on the https://github.com/ublue-os/nvidia/ repo, and probably post a message to the README and/or archive the repo.

**Comment by castrojo** [comment]
iirc archiving the repo will turn off the actions in one step.

**Comment by bsherman** [comment]
> iirc archiving the repo will turn off the actions in one step. 

Yes but do we want a message on the readme or no?

**Comment by castrojo** [comment]
Yeah, a signpost would help, I can work on that. You keep going on this.

**Comment by bsherman** [comment]
> You keep going on this.

I already added `main` repo as "Admin" to the container packages (*-nvidia) so the pushes should succeed.

I think this is ready as soon as y'all approve.

**Review by castrojo — APPROVED** [review]

File: nvidia-install.sh
Comment (p5) [inline nvidia-install.sh:1]: (Unrelated)
It might be a good idea to bundle these Nvidia install and post-install scripts into the akmods image.

**Review by p5 — APPROVED** [review]
LGTM 🤞

**Review by castrojo — APPROVED** [review]

**Review by castrojo — APPROVED** [review]

**Pull request merged**